ctapipe_io_lst/tests/test_calib.py

```diff
@@ -0,0 +1,154 @@
+
+    first_caps = get_first_capacitor_for_modules(first_capacitor_id)
+    # we just different shape (N_MODULES, N_GAINS, N_PIXELS_PER_MODULE) before
+    first_caps = first_caps.reshape((N_GAINS, N_MODULES, N_PIXELS_PER_MODULE))
+    first_caps = np.swapaxes(first_caps, 0, 1)
+    assert np.all(first_caps == expected)
+
+
+def test_read_calib_file():
+    from ctapipe_io_lst.calibration import LSTR0Corrections
+
+    mon = LSTR0Corrections._read_calibration_file(test_calib_path)
+    # only one telescope in that file
+    assert mon.tel.keys() == {1, }
+
+
+def test_read_drs4_pedestal_file():
+    from ctapipe_io_lst.calibration import LSTR0Corrections, N_CAPACITORS_PIXEL, N_ROI
+
+    pedestal = LSTR0Corrections._get_drs4_pedestal_data(test_drs4_pedestal_path)
+
+    assert pedestal.shape[-1] == N_CAPACITORS_PIXEL + N_ROI
+    # check circular boundary
+    assert np.all(pedestal[..., :N_ROI] == pedestal[..., N_CAPACITORS_PIXEL:])
+
+    # check offset is applied
+    pedestal_offset = LSTR0Corrections._get_drs4_pedestal_data(
+        test_drs4_pedestal_path, offset=100,
+    )
+    assert np.all((pedestal - pedestal_offset) == 100)
+
+
+def test_read_drs_time_calibration_file():
+    from ctapipe_io_lst.calibration import LSTR0Corrections, N_GAINS, N_PIXELS
+
+    fan, fbn = LSTR0Corrections.load_drs4_time_calibration_file(test_time_calib_path)
+
+    assert fan.shape == fbn.shape
+    assert fan.shape[0] == N_GAINS
+    assert fan.shape[1] == N_PIXELS
+
+
+def test_init():
+    from ctapipe_io_lst import LSTEventSource
+    from ctapipe_io_lst.calibration import LSTR0Corrections
+
+    subarray = LSTEventSource.create_subarray(geometry_version=4)
+    r0corr = LSTR0Corrections(subarray)
+    assert r0corr.last_readout_time.keys() == {1, }
+
+
+def test_source_with_drs4_pedestal():
+    from ctapipe_io_lst import LSTEventSource
+
+    config = Config({
+        'LSTEventSource': {
+            'LSTR0Corrections': {
+                'drs4_pedestal_path': test_drs4_pedestal_path,
+            }
+        }
+    })
+
+    source = LSTEventSource(
+        input_url=test_r0_path,
+        config=config,
+    )
+    assert source.r0_r1_calibrator.drs4_pedestal_path.tel[1] == test_drs4_pedestal_path.absolute()
+
+    with source:
+        for event in source:
+            assert event.r1.tel[1].waveform is not None
+
+
+def test_source_with_calibration():
+    from ctapipe_io_lst import LSTEventSource
+
+    config = Config({
+        'LSTEventSource': {
+            'LSTR0Corrections': {
+                'drs4_pedestal_path': test_drs4_pedestal_path,
+                'calibration_path': test_calib_path,
+            }
+        }
+    })
+
+    source = LSTEventSource(
+        input_url=test_r0_path,
+        config=config,
+    )
+
+    assert source.r0_r1_calibrator.mon_data is not None
+    with source:
+        for event in source:
+            assert event.r1.tel[1].waveform is not None
```

Comment:
I just wonder ... this test is not really testing that providing a `calibration_path` really results in the source trying to apply a calibration. 

Maybe a simple way to test this would be: 
use two sources, one with `calibration_path` and one without. And then just make sure that the waveforms are not identical? So making sure that at least *something* was done?

I mean .. this does not proof that the calibration is not adding random numbers .. but it makes sure the calibration is executed as a result of providing that path no?

Reply:
The r1 waveforms are not present at all when no calibration is applied.

Reply:
Ah thanks .. I missed that